go/consensus/pricediscovery/pricediscovery.go

```diff
@@ -0,0 +1,161 @@
+// Package pricediscovery implements dynamic price discovery.
+package pricediscovery
```

Comment:
Should move interface, `staticPriceDiscovery` and `noOpPriceDiscovery` here?

Reply:
Interface cannot be moved as it would then result in a circular import which is not supported in Go. I will move `staticPriceDiscovery` and `noOpPriceDiscovery` is only used by the no-op submission manager so it should just stay there.